### DISCUSSION TITLE
[WindowsAppSdk][WASM] How to get SVG icons working on MenuFlyoutItem.Icon?

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#8122] [Q&A] [answered]
I'm trying to load SVG icons from assets to show them as .IconImage on MenuFlyoutItem.Icon (please see attached).

AFAIK this requires SvgImageSource.RasterizePixelHeight/.RasterizePixelWidth being set (at least on WinUI), which unfortunately is not supported in UNO WASM.

How could I have SVG icons on MenuFlyoutItem.Icon?

Thanks in advance

[MenuFlyoutItemIcon.zip](https://github.com/unoplatform/uno/files/8089144/MenuFlyoutItemIcon.zip)

**Comment by DierkDroth** [comment]
@jeromelaban would you have any idea on what could be done here? Thanks

**Comment by MartinZikmund** [comment]
@DierkDroth Can you please create an issue on this? I will be working on improving SVG support, so I would like to track this problem as well

**Comment by DierkDroth** [comment]
@MartinZikmund sounds good. I'll get you a repo in the next few days.

**Comment by DierkDroth** [comment]
@MartinZikmund I just tried to create a fresh new .NET6 repo as [per here](https://platform.uno/docs/articles/get-started-winui3.html) but failed.

Please advice. Thanks

<img width="675" alt="image" src="https://user-images.githubusercontent.com/26179242/180385482-b61d122a-9385-406b-b47b-2c89211bd70c.png">

**Reply by MartinZikmund** [reply]
Can you make the repro using the old Xamarin + UWP template? I am not sure why you are not seeing the WinUI .NET 6 template at all though - maybe @ebariche might know?

**Reply by DierkDroth** [reply]
@MartinZikmund I have all my stuff on .NET6 + WindowsAppSDK. I don't think it makes sense to invest in Xamarin + UWP any longer...

**Reply by MartinZikmund** [reply]
I am just waiting for the workloads to reliably work finally 😁

**Reply by DierkDroth** [reply]
@MartinZikmund understand. That's why I'm more than hesitant to try a different setup than what I got working after investing tons of hours...

**Reply by DierkDroth** [reply]
BTW just came across [this one](https://github.com/microsoft/WindowsAppSDK/issues/2767) ... not fun

**Reply by ebariche** [reply]
The default template has changed to WinUI + NET 6, please use `dotnet new unoapp`

**Reply by ebariche** [reply]
See : https://platform.uno/blog/uno-platform-4-4-wasm-threadingexception-handling-rich-animations-gamepad-apis-and-more/

**Reply by DierkDroth** [reply]
Thanks @ebariche. That worked.

I recommend updating [the documentation](https://platform.uno/docs/articles/get-started-winui3.html) to avoid further confusion.

**Comment by DierkDroth** [comment]
@MartinZikmund here is the repo

[MenuFlyoutIcon.zip](https://github.com/unoplatform/uno/files/9173553/MenuFlyoutIcon.zip)

**Comment by DierkDroth** [comment]
@jeromelaban since we're talking about documentation issues in the other thread ... could you please browse up this thread to see that the directions [here ](https://platform.uno/docs/articles/get-started-winui3.html)are wrong
```
dotnet new unoapp-winui-net6 -o MyUnoApp
```
... but rather should be
```
dotnet new unoapp -o MyUnoApp
```

**Comment by DierkDroth** [comment]
@MartinZikmund I just updated repo above to .NET7 + latest UNO bits and found that there still are no SVG icons on menu items.

I thoughts SVG now would be working as suggested e.g. [here](https://github.com/unoplatform/uno/issues/1982)

What am I missing?

**Reply by jeromelaban** [reply]
Which repo is this?

**Reply by MartinZikmund** [reply]
> @MartinZikmund I just updated repo above to .NET7 + latest UNO bits and found that there still are no SVG icons on menu items.
> 
> I thoughts SVG now would be working as suggested e.g. [here](https://github.com/unoplatform/uno/issues/1982)
> 
> What am I missing?

@DierkDroth have you added the Uno.UI/WinUI.Svg package?

**Reply by DierkDroth** [reply]
@MartinZikmund  No I haven't. Is this absolutely required for the icons on the menu items?

FWIW SVG icons had been working in general already for quite a while. That's why I'm more than hesitent to include a package just for SVG icons on menues...

**Reply by DierkDroth** [reply]
> Which repo is this?

@jeromelaban This is the MenuFlyoutIcon.zip repo from above.

**Reply by jeromelaban** [reply]
For wasm it's not supposed to be needed. ~~Can you indicate which repo this is?~~ Thanks.

**Reply by DierkDroth** [reply]
@jeromelaban thanks for clarification. Please let me know if you needed anything else to reproduce the issue.

**Reply by jeromelaban** [reply]
Thanks. The image is present in the HTML, but for some reason the content is not being relayouted properly. @MartinZikmund can you open an issue?

**Reply by MartinZikmund** [reply]
@DierkDroth Ah sorry, didn't realize it's WASM - for WASM the package is not needed. I will look into why your repro specifically is not working, but I know I recreated it in a different way in the samples app and that form of the code did work for me - https://github.com/unoplatform/uno/blob/master/src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ImageTests/SvgImageSource_Icons.xaml.cs - so it must be some kind of difference between the two samples.

**Reply by DierkDroth** [reply]
@MartinZikmund is there a separate issue where I could track the progress? Thanks

**Reply by jeromelaban** [reply]
Let's track in https://github.com/unoplatform/uno/issues/10554.

**Comment by MartinZikmund** [comment, marked as answer]
@DierkDroth https://github.com/unoplatform/uno/pull/10579 will fix the problem

**Reply by DierkDroth** [reply]
Thanks for the update @MartinZikmund

**Reply by DierkDroth** [reply]
@MartinZikmund confirmed working. Excellent! Thanks
